spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryClientAutoConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package org.springframework.cloud.client.discovery.simple;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.cloud.client.discovery.noop.NoopDiscoveryClientAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Spring Boot Auto-Configuration for Simple Properties based Discovery Client
+ *
+ * @author Biju Kunjummen
+ */
+
+@Configuration
+@ConditionalOnProperty(value = "spring.cloud.discovery.client.simple.enabled", matchIfMissing = false)
+@EnableConfigurationProperties(SimpleDiscoveryProperties.class)
+@AutoConfigureBefore(NoopDiscoveryClientAutoConfiguration.class)
```

Comment:
Oh, okay, safe to delete NoOpDiscoveryClient and helpers altogether or should I mark it as deprecated?